### PR TITLE
[CLD-7373] Always overwrite CloudSettings based on service environment when in production

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
The CWSURL  config setting can be in incorrect states for end-user deployments for various reasons. This PR removes the ability for the config to be incorrectly configured when the service environment is set to production, and instead opts to base the URL off of the serviceenvironment value. ie, if production, it'll be `https://customers.mattermost.com` otherwise, it'll be `https://portal.test.cloud.mattermost.com`. If the serviceenvironment is not production, overrides will be respected. 

This will resolve issues around trial requests having an invalid signature, because the user's production instance is pointing at our test portal.

I've left the CWSAPIURL value to still be overridden if set, because that value is used in the Cloud flow. It's value is the same as the CWSURL for self hosted, but different for Cloud deployments so must remain overrideable. The CWSAPIURL is not used in a self hosted context.

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-7373
#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by nickmisasi** [comment]
@amyblais this could be a candidate for cherry picking back to our latest ESR - although we're nearing the very end of that support so I'm not sure if we'll want to. Let me know what you think.

File: server/public/model/config.go [hunk 1]
```diff
@@ -3034,14 +3034,13 @@ type CloudSettings struct {
...
-		case ServiceEnvironmentTest, ServiceEnvironmentDev:
-			s.CWSURL = NewString(CloudSettingsDefaultCwsURLTest)
-		}
+	switch GetServiceEnvironment() {
+	case ServiceEnvironmentProduction:
+		s.CWSURL = NewString(CloudSettingsDefaultCwsURL)
+	case ServiceEnvironmentTest, ServiceEnvironmentDev:
+		s.CWSURL = NewString(CloudSettingsDefaultCwsURLTest)
 	}
+
```
Comment (lieut-data) [inline server/public/model/config.go:3043]: Would we consider "removing" the configuration altogether and doing this check closer to the code using it?

**Review by lieut-data — APPROVED** [review]
No blockers, just one thought inline here and one comment below.

> I've left the CWSAPIURL value to still be overridden if set, because that value is used in the Cloud flow. It's value is the same as the CWSURL for self hosted, but different for Cloud deployments so must remain overrideable. The CWSAPIURL is not used in a self hosted context.

Out of band for this PR, but I wonder if a `cloud` service environment would help with these sorts of cases?

**Comment by amyblais** [comment]
> this could be a candidate for cherry picking back to our latest ESR - although we're nearing the very end of that support so I'm not sure if we'll want to. Let me know what you think.

This can be backported to both v8.1 and v9.5 supported ESRs if we want to. The latest ESR v9.5 is supported until November.

File: server/public/model/config.go (hunk 1 above)
Comment (nickmisasi) [reply server/public/model/config.go:3043]: Definitely something to consider long term. Some things are in flux around Self Serve in both Cloud and Self Hosted which depending on the outcome would have rippling effects on areas of code just like your suggestion. Until those decisions shake out I'm holding off on any more significant changes

**Review by marianunez — APPROVED** [review]

**Comment by nickmisasi** [comment]
@marianunez @lieut-data I needed to adjust the logic so that the URL could be override-able when not in a `production` serviceenvironment to allow for unit tests to run. I've updated the description and title accordingly.

File: server/channels/api4/license_test.go
Comment (lieut-data) [inline server/channels/api4/license_test.go:1]: Intentional file changes?

**Review by lieut-data — APPROVED** [review]

File: server/channels/api4/license_test.go
Comment (nickmisasi) [reply server/channels/api4/license_test.go:1]: Nope, thanks!

**Review by marianunez — APPROVED** [review]

**Pull request merged**